From e0c75c3d3e3092fdc6819f38cb8eb67a33c56ade Mon Sep 17 00:00:00 2001
From: Thomas Betker <thomas.betker@rohde-schwarz.com>
Date: Sat, 11 May 2013 11:10:46 +0200
Subject: [PATCH 402/628] xilinx_emacps: Don't assert NETIF_F_FRAGLIST feature

git://github.com/Xilinx/linux-xlnx.git xilinx-v14.7
commit 95ed5c6876833f57beabed831f3473bd66de4c55

In xemacps_probe(), do not set NETIF_F_FRAGLIST in ndev->features, only
NETIF_F_SG. xemacps_start_xmit() handles skb_shinfo(skb)->nr_frags, but
not skb_shinfo(skb)->frag_list.

Signed-off-by: Thomas Betker <thomas.betker@rohde-schwarz.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Liming Wang <liming.wang@windriver.com>
---
 drivers/net/ethernet/xilinx/xilinx_emacps.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/net/ethernet/xilinx/xilinx_emacps.c b/drivers/net/ethernet/xilinx/xilinx_emacps.c
index 8e48636..9a5c863 100644
--- a/drivers/net/ethernet/xilinx/xilinx_emacps.c
+++ b/drivers/net/ethernet/xilinx/xilinx_emacps.c
@@ -2633,7 +2633,7 @@ static int xemacps_probe(struct platform_device *pdev)
 	ndev->watchdog_timeo = TX_TIMEOUT;
 	ndev->ethtool_ops = &xemacps_ethtool_ops;
 	ndev->base_addr = r_mem->start;
-	ndev->features = NETIF_F_IP_CSUM | NETIF_F_FRAGLIST | NETIF_F_SG;
+	ndev->features = NETIF_F_IP_CSUM | NETIF_F_SG;
 	netif_napi_add(ndev, &lp->napi, xemacps_rx_poll, XEMACPS_NAPI_WEIGHT);
 
 	lp->ip_summed = CHECKSUM_UNNECESSARY;
-- 
1.7.5.4

